pkg/types/ibmcloud/validation/machinepool.go

```diff
@@ -25,3 +42,40 @@ func validateBootVolume(bv *ibmcloud.BootVolume, path *field.Path) field.ErrorLi
 	}
 	return allErrs
 }
+
+func validateDedicatedHosts(dhosts []ibmcloud.DedicatedHost, itype string, zones []string, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Length of dedicated hosts must match platform zones
+	if len(dhosts) != len(zones) {
```

Comment:
Why do we require the user to specify the zones in two different locations in the exact same order? Can't we use the zones defined at the machine-pool level and omit the zones from the dedicated hosts?

Reply:
Yes, we can. I didn't know if it was an anti-pattern to make that sort of linkage between 2 params.